File: plugins/fr.cea.nabla.javalib/src/linearalgebrajava/Matrix.java
/*******************************************************************************
 * Copyright (c) 2020 CEA
 * This program and the accompanying materials are made available under the 
 * terms of the Eclipse Public License 2.0 which is available at
 * http://www.eclipse.org/legal/epl-2.0.
 *
 * SPDX-License-Identifier: EPL-2.0
 * Contributors: see AUTHORS file
 *******************************************************************************/
package linearalgebrajava;

import org.apache.commons.math3.linear.AbstractRealMatrix;
import org.apache.commons.math3.linear.Array2DRowRealMatrix;
//import org.apache.commons.math3.linear.OpenMapRealMatrix;

public class Matrix
{
	private final Object lock = new Object();
	private final AbstractRealMatrix data;
	private final String name;

	public Matrix(final String name, final int nbRows, final int nbCols)
	{
		this.name = name;
		// Dense
		this.data = new Array2DRowRealMatrix(nbRows, nbCols);
		// Sparse
		// this.data = new OpenMapRealMatrix(nbRows, nbCols)
	}

	public String getName()
	{
		return name;
	}

	public AbstractRealMatrix getData()
	{
		return data; 
	}

	int getNbRows()
	{
		return data.getRowDimension();
	}

	int getNbCols()
	{
		return data.getColumnDimension();
	}

	public double getValue(int i, int j)
	{
		return data.getEntry(i, j); 
	}

	public void setValue(int i, int j, double value)
	{
		synchronized(lock) { data.setEntry(i, j, value); }
	}

	@Override
	public String toString()
	{
		StringBuilder sb = new StringBuilder("");
		for (int i = 0; i < data.getColumnDimension(); i++)
			for (int j = 0; j < data.getRowDimension(); j ++)
			{
				sb.append(data.getEntry(i, j));
				sb.append(" ");
			}
		return sb.toString();
	}
}

File: plugins/fr.cea.nabla.pythonlib/src/linearalgebra/linearalgebranumpy/matrix.py
"""
 *******************************************************************************/
 * Copyright (c) 2022 CEA
 * This program and the accompanying materials are made available under the 
 * terms of the Eclipse Public License 2.0 which is available at
 * http://www.eclipse.org/legal/epl-2.0.
 *
 * SPDX-License-Identifier: EPL-2.0
 * Contributors: see AUTHORS file
 *******************************************************************************/
"""
import numpy as np

class Matrix:
	def __init__(self, name, data):
		self.__name = name
		self.__data = data

	@classmethod
	def zeros(cls, name, nbRows, nbCols):
		return cls(name, np.zeros((nbRows, nbCols), dtype=np.double))

	def getName(self):
		return self.__name

	def getNbRows(self):
		return self.__data.shape[0]

	def getNbCols(self):
		return self.__data.shape[1]

	def getValue(self, i, j):
		return self.__data[i, j]

	def setValue(self, i, j, value):
		self.__data[i, j] = value

	def getData(self):
		return self.__data
